kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ImageInstanceDescriptor.java

```diff
@@ -0,0 +1,200 @@
+     * @return
+     */
+    public long getImageSize() {
+        return imageSize;
+    }
+
+    /**
+     * Returns all of a Image's tags as a Map<String,String>.
+     * @return
+     */
+    public Map<String, String> getImageTags() {
+        return imageLabels;
+    }
+
+    /**
+     * Creates a builder for creating a new {@link ImageInstanceDescriptor}
+     * instance.
+     *
+     * @return the builder.
+     */
+    public static ImageInstanceDescriptorBuilder builder() {
+        return new ImageInstanceDescriptorBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.imageName, this.imageTag, this.imageId, this.imageAuthor, this.imageArch,
+                this.imageSize, this.imageLabels);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof ImageInstanceDescriptor)) {
+            return false;
+        }
+        ImageInstanceDescriptor other = (ImageInstanceDescriptor) obj;
+        return Objects.equals(this.imageName, other.imageName) && Objects.equals(this.imageTag, other.imageTag)
+                && Objects.equals(this.imageId, other.imageId) && Objects.equals(this.imageAuthor, other.imageAuthor)
+                && Objects.equals(this.imageArch, other.imageArch) && Objects.equals(this.imageSize, other.imageSize)
+                && Objects.equals(this.imageLabels, other.imageLabels);
+    }
+
+    public static final class ImageInstanceDescriptorBuilder {
+
+        private String imageName;
+        private String imageTag;
+        private String imageId;
+        private String imageAuthor;
+        private String imageArch;
+        private long imageSize;
+        private Map<String, String> imageLabels;
+
+        public ImageInstanceDescriptorBuilder setImageName(String imageName) {
+            this.imageName = nullToEmpty(imageName);
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setImageTag(String imageTag) {
+            this.imageTag = nullToEmpty(imageTag);
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setImageId(String imageId) {
+            this.imageId = nullToEmpty(imageId);
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setImageAuthor(String imageAuthor) {
+            this.imageAuthor = nullToEmpty(imageAuthor);
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setImageArch(String imageArch) {
+            this.imageArch = nullToEmpty(imageArch);
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setimageSize(long imageSize) {
+            this.imageSize = imageSize;
+            return this;
+        }
+
+        public ImageInstanceDescriptorBuilder setImageLabels(Map<String, String> imageLabels) {
+            this.imageLabels = imageLabels;
```

Comment:
what about if this field is null?

Reply:
The only field that cannot be null from my research is the Image Id. I have modified it accordingly.